codegen/ace/src/layout.rs

```diff
@@ -89,19 +95,25 @@ impl Layout {
         }
 
         fn align(offset: &mut usize, alignment: Alignment) {
-            *offset += offset.next_multiple_of(alignment as usize);
+            *offset = offset.next_multiple_of(alignment as usize);
         }
 
         // The arrays of all public inputs are stored contiguously.
         let public_inputs: BTreeMap<_, _> = air
             .public_inputs
             .iter()
-            .map(|(ident, pi)| (*ident, next_region(offset, pi.size(), Alignment::Element)))
+            .filter_map(|(ident, pi)| {
+                if let PublicInput::Vector { .. } = pi {
+                    Some((*ident, allocate_region(offset, pi.size(), Alignment::Element)))
+                } else {
+                    None
+                }
+            })
             .collect();
 
-        // Ensure the entire region containing the public inputs is double-word aligned
-        // since it is hashed as one contiguous array.
-        align(offset, Alignment::DoubleWord);
+        // Ensure the entire region containing the public inputs is double-double-word aligned
+        // since it is hashed as one contiguous array and processed in batches of 8 inputs.
+        align(offset, Alignment::DoubleDoubleWord);
```

Comment:
nit: would something like `QuadWord` be better than `DoubleDoubleWord`?

Reply:
Good point, changed